zebra/label_manager.c

```diff
@@ -138,7 +136,11 @@ assign_label_chunk (label_owner_t owner)
 		else
 				lmc->start = ((struct label_manager_chunk *)
 							  listgetdata(listtail(lbl_mgr.lc_list)))->end + 1;
-		lmc->end = lmc->start + lbl_mgr.chunk_size - 1;
+		if (lmc->start > MPLS_MAX_UNRESERVED_LABEL - size + 1) {
```

Comment:
I would prefer writing the test this way for readability:

if (lmc->start + size - 1 > MPLS_MAX_UNRESERVED_LABEL)